src/bootstrap/builder.rs

```diff
@@ -303,15 +303,14 @@ impl StepDescription {
     }
 
     fn maybe_run(&self, builder: &Builder<'_>, pathsets: Vec<PathSet>) {
-        if pathsets.iter().any(|set| self.is_excluded(builder, set)) {
-            return;
-        }
+        let excluded_pathsets: Vec<PathSet> = 
+            pathsets.iter().filter(|set| !self.is_excluded(builder, set)).cloned().collect();
```

Comment:
```suggestion
        pathsets.retain(|set| !self.is_excluded(builder, set));
```
Avoids a duplication of the vector

Reply:
Thank you and it's very helpful